stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClientImpl.kt

```diff
@@ -112,13 +112,13 @@ internal class ChatClientImpl(
         connectionListener = listener
         config.isAnonymous = true
         warmUp()
-        notifications.onSetUser()
         getTokenAndConnect {
             socket.connectAnonymously()
         }
     }
 
     override fun getGuestToken(userId: String, userName: String): Call<GuestUser> {
+        config.isAnonymous = true
```

Comment:
Why do we need to do this side-effect here?

Reply:
Maybe it's better to have `isAnonymous` true by default in ChatClientConfig. This request should be `anonymous` (`stream-auth-type` set to `anonymous` in `HeadersInterceptor`)

Reply:
What I mean is that we shouldn't mutate the status of `config.isAnonymous` inside of a method that says `getGuestToken()`, because this method is doing more stuff that it said to do.
It should be done outside